### PR TITLE
Drop PostgreSQL 9.4 support

### PR DESCRIPTION
**Pull request by mtuncer**
Removed support for pre-9.5 versions. Now Citus will only support PostgreSQL 9.5.x builds.

- removed conditional compile flags that enabled 9.4 builds
- changed configure.in file to accept only 9.5 version
- removed 9.4 build from travis file
- removed 9.4 specific citus_readfuncs94.c and ruleutils94.c files
- remove alternate test expected file for multi_upsert.

### CODE REVIEW

File: configure.in [hunk 1]
```diff
@@ -36,7 +36,7 @@ if test -z "$version_num"; then
   AC_MSG_ERROR([Could not detect PostgreSQL version from pg_config.])
 fi
 
-if test "$version_num" != '9.4' -a "$version_num" != '9.5'; then
+if "$version_num" != '9.5'; then
```
Comment (anarazel) [inline configure.in]: Uh, that doesn't look right. You removed the actual test.

File: src/backend/distributed/executor/multi_utility.c [hunk 2]
```diff
@@ -1595,19 +1573,15 @@ ReplicateGrantStmt(Node *parsetree)
...
 		RoleSpec *spec = lfirst(granteeCell);
-#else
-		PrivGrantee *spec = lfirst(granteeCell);
-#endif
 
 		if (!isFirst)
 		{
 			appendStringInfoString(&granteesString, ", ");
 		}
+
```
Comment (anarazel) [inline src/backend/distributed/executor/multi_utility.c]: spurious change.

File: src/backend/distributed/test/prune_shard_list.c [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 #include <string.h>
 
-#if (PG_VERSION_NUM >= 90500 && PG_VERSION_NUM < 90600)
+#if (PG_VERSION_NUM < 90600)
```
Comment (anarazel) [inline src/backend/distributed/test/prune_shard_list.c]: Hm, whydid you leave that 90600 part of the test here? This looks odd.

File: src/bin/csql/large_obj.c [hunk 4]
```diff
@@ -7,18 +7,12 @@
  */
 #include "postgres_fe.h"
 #include "large_obj.h"
-
```
Comment (anarazel) [inline src/bin/csql/large_obj.c]: Unrelated.

File: src/backend/distributed/executor/multi_utility.c (hunk 2 above)
Comment (mtuncer) [inline src/backend/distributed/executor/multi_utility.c]: that was to follow "empty line after the code block" convention. But is it not related to 9.4 here.

File: src/backend/distributed/test/prune_shard_list.c (hunk 3 above)
Comment (mtuncer) [inline src/backend/distributed/test/prune_shard_list.c]: oddness started with the check ```PG_VERSION_NUM >= 90500 && PG_VERSION_NUM < 90600```. I do not know why we had that in the first place. removing it.

File: configure.in (hunk 1 above)
Comment (mtuncer) [inline configure.in]: that was why I wanted to wait for travis to run. correcting it.

File: src/backend/distributed/planner/multi_master_planner.c [hunk 5]
```diff
@@ -167,15 +167,10 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 	}
 
 	/* finally create the plan */
-#if (PG_VERSION_NUM >= 90500)
+
```
Comment (anarazel) [inline src/backend/distributed/planner/multi_master_planner.c]: What's that newline for?

Comment (mtuncer) [inline src/backend/distributed/planner/multi_master_planner.c]: gone

**Comment by anarazel** [comment]
Looks good to me!

**Comment by mtuncer** [comment]
rebased, squashed, verified travis is fine.
merging
